torch/package/package_importer.py

```diff
@@ -317,6 +330,10 @@ def _make_module(
         ns["__builtins__"] = self.patched_builtins
         ns["__torch_package__"] = True
 
+        if is_package and name in self.selective_extern_packages:
```

Comment:
We shouldn't be referencing `self.selective_extern_packages` here. The reason is that it makes the dataflow confusing, now in order to understand what to do when making a module you are checking two sources of truth with overlapping responsibility (the module tree and the selective intern list)

Previously the flow was: on initialization PackageImporter builds up the module tree with all the necessary information to service imports. We should preserve that property and not check a side channel for information on how to handle a particular modules.